### PR TITLE
Adding tags to interest rate schema

### PR DESCRIPTION
**Pull request by phorical**
## Description

Adding a tags field to interest rates schema. This is currently used to tag SCRA rates but functionality could be expanded in the future (e.g. adding notes about promos, etc)

> Give your teammates a high level overview of what's happening. What changes are being made? What is in scope or out of scope?

## Testing

> Teach your teammates how to run and check that your work does what you say it does.

- [ ] Is QA required?

## Related issues

> Help your teammates keep track of work. Link relevant PRs and tickets.
>
> You can link Linear tickets by ID within the title or add the entire Linear ticket URL inside the description. GH integrations will link tickets and PRs.

### CODE REVIEW

**Review by michael-canopy — APPROVED** [review]

File: schemas/InterestRate.json [hunk 1]
```diff
@@ -24,6 +24,10 @@
         "type": "number",
         "description": "The interest rate percent effective during this window",
         "example": "1.23"
+      },
+      "tags": {
```
Comment (ps2-controller) [inline schemas/InterestRate.json:28]: Object needs `properties` attribute. 
I need a bit of a refresher for what we want to accomplish with `tags` --- If we're using this to allow arbitrary additional properties, we should follow the `external_fields` pattern used elsewhere in the API

**Review by ps2-controller — CHANGES_REQUESTED** [review]

File: schemas/InterestRate.json (hunk 1 above)
Comment (michael-canopy) [reply schemas/InterestRate.json:28]: Ooh that's right. I'm a little rusty on OpenAPI specs nowadays...

Comment (ps2-controller) [reply schemas/InterestRate.json:28]: All good! I'm surprised it built actually, maybe it's not _technically_ required, but practically speaking it is

Comment (phorical) [reply schemas/InterestRate.json:28]: We had talked about this a bit @ps2-controller as a way to generalize the need to have an account/interest rate somehow marked as coming from SCRA. This allows further expansion into use cases like notes on the interest rate or denoting specific promos that apply. I can add the `scra` properties here or we could talk about refactoring this to something else

**Review by ps2-controller — APPROVED** [review]

**Comment by ps2-controller** [comment]
Approved, but qq -- does inclusion of this tag affect system behavior in any way? If so, that'd be good to document in its description.

**Comment by phorical** [comment]
@ps2-controller The scra_start and scra_end functionality uses this tag under the hood. That being said, it can be expanded beyond just SCRA for future use cases. I am not expecting our customers to include any tags for now; I expect them to use the scra_start and _end functions which will create the `{"scra": true}` tag for them on the interest rate.

**Comment by ps2-controller** [comment]
Gotcha, makes sense!
One last note --- should these tags print booleans, or nested information? I think with SCRA, there's an "expected" end date that doesn't actually map to the intentional system end date, which is infinity, right? Would this be a good place to include that extra SCRA-specific static information? 

i.e. 
{
  "scra": {
       "start_date": "jan 2",
       "expected_end_date": "Aug 2",
       "other_scra_foo": "bar"
   }
}

**Comment by phorical** [comment]
I like where you're going with that, and I think that's a good idea but probably a Day 2 thing. I would love to get the functionality out into the open for testing/BHG first, then expand on top of it with additional functionality/improvements.

**Comment by ps2-controller** [comment]
Sounds good!

**Pull request merged**